### PR TITLE
Conform to OT tagging for databases and message brokers

### PR DESCRIPTION
**Pull request by sm7x**

### CODE REVIEW

File: dd-java-agent/instrumentation/couchbase-2.0/src/main/java/datadog/trace/instrumentation/couchbase/client/CouchbaseBucketInstrumentation.java [hunk 1]
```diff
@@ -118,8 +119,8 @@ public void call() {
           GlobalTracer.get()
               .buildSpan("couchbase.call")
               .withTag(DDTags.RESOURCE_NAME, resourceName)
-              .withTag("bucket", bucket)
               .start();
+      Tags.DB_INSTANCE.set(span, bucket);
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/couchbase-2.0/src/main/java/datadog/trace/instrumentation/couchbase/client/CouchbaseBucketInstrumentation.java]: any reason not to just update `"bucket"` with the db instance key accessor?

Comment (sm7x) [reply dd-java-agent/instrumentation/couchbase-2.0/src/main/java/datadog/trace/instrumentation/couchbase/client/CouchbaseBucketInstrumentation.java]: I'm not really following

File: dd-java-agent/instrumentation/kafka-streams-0.11/src/test/groovy/KafkaStreamsTest.groovy [hunk 2]
```diff
@@ -115,7 +115,6 @@ class KafkaStreamsTest extends AgentTestRunner {
     def received = records.poll(10, TimeUnit.SECONDS)
     received.value() == greeting.toLowerCase()
     received.key() == null
-
     assertTraces(3) {
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/kafka-streams-0.11/src/test/groovy/KafkaStreamsTest.groovy]: nit: whitespace

File: dd-java-agent/instrumentation/lettuce-5/src/main/java8/datadog/trace/instrumentation/lettuce/LettuceClientDecorator.java [hunk 3]
```diff
@@ -44,6 +44,10 @@ protected String dbUser(final RedisURI connection) {
 
   @Override
   protected String dbInstance(final RedisURI connection) {
+    return null;
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/lettuce-5/src/main/java8/datadog/trace/instrumentation/lettuce/LettuceClientDecorator.java]: why wouldn't this be `connection.getDatabase()`?

File: dd-java-agent/instrumentation/datastax-cassandra-3/src/main/java/datadog/trace/instrumentation/datastax/cassandra/CassandraClientDecorator.java [hunk 4]
```diff
@@ -46,7 +46,7 @@ protected String dbUser(final Session session) {
 
   @Override
   protected String dbInstance(final Session session) {
-    return session.getLoggedKeyspace();
+    return null;
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/datastax-cassandra-3/src/main/java/datadog/trace/instrumentation/datastax/cassandra/CassandraClientDecorator.java]: what's wrong with using the keyspace?

File: dd-java-agent/instrumentation/lettuce-5/src/main/java8/datadog/trace/instrumentation/lettuce/LettuceClientDecorator.java (hunk 3 above)
Comment (sm7x) [reply dd-java-agent/instrumentation/lettuce-5/src/main/java8/datadog/trace/instrumentation/lettuce/LettuceClientDecorator.java]: It returns a number, an integer. Is this (a number) what is preferred?

Comment (rmfitzpatrick) [reply dd-java-agent/instrumentation/lettuce-5/src/main/java8/datadog/trace/instrumentation/lettuce/LettuceClientDecorator.java]: Yes, the initial/default connection is for db 0 and people are able to configure more as their application, deployment, and upgrade patterns require.

File: dd-java-agent/instrumentation/elasticsearch/transport-5.3/src/main/java/datadog/trace/instrumentation/elasticsearch5_3/TransportActionListener.java [hunk 5]
```diff
@@ -34,7 +34,7 @@ private void onRequest(final ActionRequest request) {
     if (request instanceof IndicesRequest) {
       final IndicesRequest req = (IndicesRequest) request;
       if (req.indices() != null) {
-        span.setTag("elasticsearch.request.indices", Joiner.on(",").join(req.indices()));
+        Tags.DB_INSTANCE.set(span, Joiner.on(",").join(req.indices()));
```
Comment (mpetazzoni) [inline dd-java-agent/instrumentation/elasticsearch/transport-5.3/src/main/java/datadog/trace/instrumentation/elasticsearch5_3/TransportActionListener.java]: I'm not sure this one is appropriate, especially when people use time-based or date-based indices. Elasticsearch typically will have a cluster name that is more appropriate to use for the `db.instance` tag.

File: dd-java-agent/instrumentation/elasticsearch/transport-6/src/main/java/datadog/trace/instrumentation/elasticsearch6/TransportActionListener.java [hunk 6]
```diff
@@ -38,7 +38,7 @@ private void onRequest(final ActionRequest request) {
     if (request instanceof IndicesRequest) {
       final IndicesRequest req = (IndicesRequest) request;
       if (req.indices() != null) {
-        span.setTag("elasticsearch.request.indices", Joiner.on(",").join(req.indices()));
+        Tags.DB_INSTANCE.set(span, Joiner.on(",").join(req.indices()));
```
Comment (mpetazzoni) [inline dd-java-agent/instrumentation/elasticsearch/transport-6/src/main/java/datadog/trace/instrumentation/elasticsearch6/TransportActionListener.java]: Ditto

File: dd-java-agent/instrumentation/jetty-8/jetty-8.gradle [hunk 7]
```diff
@@ -35,7 +35,7 @@ dependencies {
   testCompile group: 'org.eclipse.jetty', name: 'jetty-servlet', version: '8.0.0.v20110901'
   testCompile group: 'org.eclipse.jetty', name: 'jetty-continuation', version: '8.0.0.v20110901'
 
-  latestDepTestCompile group: 'org.eclipse.jetty', name: 'jetty-server', version: '+'
-  latestDepTestCompile group: 'org.eclipse.jetty', name: 'jetty-servlet', version: '+'
-  latestDepTestCompile group: 'org.eclipse.jetty', name: 'jetty-continuation', version: '+'
+  latestDepTestCompile group: 'org.eclipse.jetty', name: 'jetty-server', version: '9.+'
```
Comment (mpetazzoni) [inline dd-java-agent/instrumentation/jetty-8/jetty-8.gradle]: Hmm, this is in jetty-8, are we sure that's the right version spec?

File: dd-java-agent/instrumentation/spymemcached-2.12/src/test/groovy/datadog/trace/instrumentation/spymemcached/SpymemcachedTest.groovy [hunk 8]
```diff
@@ -635,7 +635,6 @@ class SpymemcachedTest extends AgentTestRunner {
         "${Tags.COMPONENT.key}" COMPONENT_NAME
         "${Tags.SPAN_KIND.key}" Tags.SPAN_KIND_CLIENT
         "${Tags.DB_TYPE.key}" CompletionListener.DB_TYPE
-        "$Tags.DB_INSTANCE.key" ~/Connection Status \{ \w*\/127.0.0.1:\d+ active: true, authed: true, last read: \d+ ms ago }/
```
Comment (mpetazzoni) [inline dd-java-agent/instrumentation/spymemcached-2.12/src/test/groovy/datadog/trace/instrumentation/spymemcached/SpymemcachedTest.groovy]: So what `db.instance` do they get?

Comment (sm7x) [reply dd-java-agent/instrumentation/spymemcached-2.12/src/test/groovy/datadog/trace/instrumentation/spymemcached/SpymemcachedTest.groovy]: I haven't found any information that would qualify as a `db.instance` yet.

File: dd-java-agent/instrumentation/jetty-8/jetty-8.gradle (hunk 7 above)
Comment (rmfitzpatrick) [reply dd-java-agent/instrumentation/jetty-8/jetty-8.gradle]: This instruments jetty 8+, but it looks like the recently released 10 alpha is incompatible and will require updates, possibly to the build system.

**Pull request merged**